spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java

```diff
@@ -75,20 +81,79 @@ public void setRetryTemplate(RetryTemplate retryTemplate) {
 	 * @return queryable store.
 	 */
 	public <T> T retrieveQueryableStore(String storeName, QueryableStoreType<T> storeType) {
+		primeKafkaStreamsObject();
+		StoreQueryParameters<T> storeQueryParams = StoreQueryParameters.fromNameAndType(storeName, storeType);
+
+		return this.retryTemplate.execute(context -> {
+			try {
+				return this.kafkaStreams.store(storeQueryParams);
+			}
+			catch (Exception e) {
+				throw new IllegalStateException("Error retrieving state store: " + storeName, e);
+			}
+		});
+	}
+
+	private void primeKafkaStreamsObject() {
 		if (this.kafkaStreams == null) {
 			this.kafkaStreams = this.streamsBuilderFactoryBean.getKafkaStreams();
 		}
 		Assert.notNull(this.kafkaStreams, "KafkaStreams cannot be null. " +
 				"Make sure that the corresponding StreamsBuilderFactoryBean has started properly.");
-		StoreQueryParameters<T> storeQueryParams = StoreQueryParameters.fromNameAndType(storeName, storeType);
+	}
+
+	/**
+	 * Gets the current {@link HostInfo} where this Kafka Streams application is running on.
```

Comment:
Method Javadocs must be imperative.
Kinda we give a command to do from the first verb.